controllers/configuration/backend/backend.go

```diff
@@ -53,148 +43,124 @@ type Backend interface {
+	}
+	return initFunc(k8sClient, backendConf, credentials)
 }
 
-func handleInlineBackendHCL(code string, k8sClient client.Client) (Backend, error) {
-
-	type BackendConfigWrap struct {
-		Backend ParsedBackendConfig `hcl:"backend,block"`
+func handleDefaultBackend(configuration *v1beta2.Configuration, k8sClient client.Client) (Backend, error) {
+	if configuration.Spec.Backend != nil {
+		if configuration.Spec.Backend.SecretSuffix == "" {
+			configuration.Spec.Backend.SecretSuffix = configuration.Name
+		}
+		configuration.Spec.Backend.InClusterConfig = true
+	} else {
+		configuration.Spec.Backend = &v1beta2.Backend{
+			SecretSuffix:    configuration.Name,
+			InClusterConfig: true,
+		}
 	}
+	return newDefaultK8SBackend(configuration.Spec.Backend.SecretSuffix, k8sClient), nil
+}
 
+func handleInlineBackendHCL(hclCode string) (string, interface{}, error) {
 	type TerraformConfig struct {
-		Remain    interface{} `hcl:",remain"`
 		Terraform struct {
-			Remain  interface{}         `hcl:",remain"`
-			Backend ParsedBackendConfig `hcl:"backend,block"`
+			Backend struct {
+				Name  string   `hcl:"name,label"`
+				Attrs hcl.Body `hcl:",remain"`
+			} `hcl:"backend,block"`
 		} `hcl:"terraform,block"`
 	}
 
-	hclFile, diags := hclparse.NewParser().ParseHCL([]byte(code), "backend")
+	hclFile, diags := hclparse.NewParser().ParseHCL([]byte(hclCode), "backend")
 	if diags.HasErrors() {
-		return nil, fmt.Errorf("there are syntax errors in the inline backend hcl code: %w", diags)
+		return "", nil, fmt.Errorf("there are syntax errors in the inline backend hcl code: %w", diags)
 	}
 
-	// try to parse hclFile to Config or BackendConfig
+	// try to parse hclFile to TerraformConfig or TerraformConfig.Terraform
 	config := &TerraformConfig{}
-	// nolint:staticcheck
-	backendConfig := &ParsedBackendConfig{}
 	diags = gohcl.DecodeBody(hclFile.Body, nil, config)
 	if diags.HasErrors() || config.Terraform.Backend.Name == "" {
-		backendConfigWrap := &BackendConfigWrap{}
-		diags = gohcl.DecodeBody(hclFile.Body, nil, backendConfigWrap)
-		if diags.HasErrors() || backendConfigWrap.Backend.Name == "" {
-			return nil, fmt.Errorf("the inline backend hcl code is not valid Terraform backend configuration: %w", diags)
+		diags = gohcl.DecodeBody(hclFile.Body, nil, &config.Terraform)
+		if diags.HasErrors() || config.Terraform.Backend.Name == "" {
+			return "", nil, fmt.Errorf("the inline backend hcl code is not valid Terraform backend configuration: %w", diags)
 		}
-		backendConfig = &backendConfigWrap.Backend
-	} else {
-		backendConfig = &config.Terraform.Backend
 	}
 
-	initFunc := backendInitFuncMap[backendConfig.Name]
-	if initFunc == nil || initFunc.initFuncFromHCL == nil {
-		return nil, fmt.Errorf("backend type (%s) is not supported", backendConfig.Name)
+	backendType := config.Terraform.Backend.Name
+
+	var backendConf interface{}
+	switch strings.ToLower(backendType) {
+	case "kubernetes":
```

Comment:
Convert these backends to constants please as they are used multiple times.

Reply:
OK